core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixTextToSpeech.java

```diff
@@ -131,8 +131,26 @@ public void readAloud(WebView webView) {
           context.getResources().getString(R.string.tts_lang_not_supported),
           Toast.LENGTH_LONG).show();
       } else {
+        // check if voice needs to be installed for API below 21
+        if(VERSION.SDK_INT < VERSION_CODES.LOLLIPOP
+          && tts.getFeatures(locale).contains(TextToSpeech.Engine.KEY_FEATURE_NOT_INSTALLED)){
+          Toast.makeText(context,
+            context.getResources().getString(R.string.tts_lang_not_supported),
+            Toast.LENGTH_LONG).show();
+          return;
+        }
+
         tts.setLanguage(locale);
 
+        // check if voice needs to be installed for API 21 or higher
+        if (VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP
+          && tts.getVoice().getFeatures().contains(TextToSpeech.Engine.KEY_FEATURE_NOT_INSTALLED)){
+          Toast.makeText(context,
+            context.getResources().getString(R.string.tts_lang_not_supported),
```

Comment:
couple of things here
1. Toast.makeText has a method that takes stringIds
1. context has a `getString` method so you don't have to use resources
1. Sadly this is a java file so it won't look as nice but use `ContextExtensionsKt.toast` for showing toasts

Reply:
Great points, I will fix them on the next pull request. This file, however, uses toasts in the exact same way in other code parts, should I do something about it ?

Reply:
It is more an issue of the places that uses toasts are not in Kotlin yet, in kotlin it would be the much nicer `context.toast(R.string.someString)`.We just shouldn't add more examples of it the old way.
I would think that when we convert the files the static function call will be automatically converted to the extension function.
If you want to find some file and convert it to kotlin I certainly won't complain